File: Interview_Prep__App/native_java/src/main/java/com/dreams/androidquizapp/network/OkHttpProvider.java
package com.dreams.androidquizapp.network;

import okhttp3.OkHttpClient;
import okhttp3.OkHttpClient.Builder;
import okhttp3.logging.HttpLoggingInterceptor;
import okhttp3.logging.HttpLoggingInterceptor.Level;

import static java.util.concurrent.TimeUnit.SECONDS;

public class OkHttpProvider{
    private static final int WRITE_TIMEOUT = 60;
    private static final int READ_TIMEOUT = 180;
    private static final int CONNECT_TIMEOUT = 90;

    public OkHttpClient getClient() {
        Builder okBuilder = new Builder()
                .writeTimeout(WRITE_TIMEOUT, SECONDS)
                .readTimeout(READ_TIMEOUT, SECONDS)
                .connectTimeout(CONNECT_TIMEOUT, SECONDS);

        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        loggingInterceptor.setLevel(Level.BODY);

        return okBuilder.build();
    }
}

File: Interview_Prep_App/native_kotlin/retrofit2_moshi_Androidlion48/src/main/java/com/interviewprep/moshi_androidlion48/network/OkHttpProvider.kt
package com.interviewprep.moshi_androidlion48.network

import okhttp3.OkHttpClient
import okhttp3.OkHttpClient.Builder
import java.util.concurrent.TimeUnit.SECONDS

class OkHttpProvider {
    val client: OkHttpClient
        get() {
            val okBuilder: Builder = Builder()
                .writeTimeout(WRITE_TIMEOUT.toLong(), SECONDS)
                .readTimeout(READ_TIMEOUT.toLong(), SECONDS)
                .connectTimeout(CONNECT_TIMEOUT.toLong(), SECONDS)
            return okBuilder.build()
        }
    // TODO Optional: Create Interceptors

    companion object {

        // Constant variables for timeouts.
        private const val WRITE_TIMEOUT = 60
        private const val READ_TIMEOUT = 180
        private const val CONNECT_TIMEOUT = 90
    }
}
